tests/store/test_obssurface.py

```diff
@@ -1020,3 +1055,33 @@ def test_optional_metadata():
     rgl_ch4_metadata = rgl_ch4.metadata
 
     assert "project" in rgl_ch4_metadata
+
+
+@pytest.mark.parametrize("filepath, site, instrument, sampling_period, network, inlet, store, source_format, update_mismatch",
```

Comment:
Great to use parameterisation in these kinds of cases to make it easier to check out multiple scenarios but it can end up with a lot of inputs as you've found here.

One small suggestion to reduce the length of these inputs a little would be to just include a "filename" and create the filepath within the function using the source_format which is also passed. You could also just specify the store within the test as "user" is being used when getting the bucket e.g.
```
@pytest.mark.parametrize("filename, site, instrument, sampling_period, network, inlet, source_format, update_mismatch",
                         [
                             ("DECC-picarro_TAC_20130131_co2-185m-20220928.nc",
                              "tac", "picarro", "1h", "decc", "185m", "openghg", "from_definition"),

                             ("ch4_bao_tower-insitu_1_ccgg_all.nc",
                              "bao", None, None, "noaa", None, "noaa", "from_source"),

                             ("ICOS_ATC_L2_L2-2024.1_RGL_90.0_CTS.CH4",
                              "rgl", "g2301", None, "icos", None, "icos", "never")
                         ])
def test_sync_surface_metadata(filename, site, instrument, sampling_period, network, inlet,
                               source_format, update_mismatch):
    """
    Test metadata and attribues are correctly synced.
    """
    clear_test_stores()

    store = "user"
    bucket = get_writable_bucket(name=store)

    filepath = get_surface_datapath(filename, source_format)

...
```